hooks/useStakingData.ts

```diff
@@ -61,10 +59,12 @@ const useStakingData = () => {
 		contractInterface: stakingRewardsABI,
 	};
 
-	const rewardEscrowContract = {
-		addressOrName: REWARD_ESCROW[network?.id],
-		contractInterface: rewardEscrowABI,
-	};
+	const rewardEscrowContract = useMemo(() => {
+		return {
+			addressOrName: REWARD_ESCROW[network?.id],
+			contractInterface: rewardEscrowABI,
+		};
+	}, [network?.id]);
```

Comment:
Should this be done for all the contracts?

Reply:
This contract is the only `deps` in other hooks so we agreed to not make changes for others after the discord chats.